### ISSUE TITLE
Story locking doesn't work for contributors

### ISSUE THREAD

**Issue by bmattb** [swissspidy/web-stories-wp#11047]
## Bug Description
Originally reported by @BrittanyIRL 
The lock icon doesn’t show up in ‘all stories’ page if I’m logged in as a contributor 

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

**Comment by swissspidy** [comment]
This is working as intended.

Contributors are only allowed to edit their own stories, not someone else's. So if someone else is editing a story, there's no point in showing it to contributors, because they cannot edit.

But if someone else is editing a contributor's story, the contributor user does see it as expected:

<img width="696" alt="Screenshot 2022-03-23 at 14 40 32" src="https://user-images.githubusercontent.com/841956/159713204-f021526a-4870-46cf-bc9c-038fdfbbb59e.png">

<img width="506" alt="Screenshot 2022-03-23 at 14 42 25" src="https://user-images.githubusercontent.com/841956/159713343-29896fbe-c584-4164-97ea-358e3831a667.png">
<img width="249" alt="Screenshot 2022-03-23 at 14 42 17" src="https://user-images.githubusercontent.com/841956/159713351-7b0aa8d1-b8ce-467c-9fe2-6edccfc8cba4.png">


In short, you only see the story locking info if you have permission to actually edit the story.
